### ISSUE TITLE
fts(3), as used in base system, breaks on directories that do not contain `..' entry

### ISSUE THREAD

**Issue by ghost** [elad/gnats-nb#387]
### Metadata
- Number: [16651](http://gnats.netbsd.org/16651)
- Originator: Sergey Svishchev &lt;[svs@ropnet.ru](mailto:svs@ropnet.ru)&gt;
- Arrived: Sat May 04 01:57:00 +0000 2002
- Last modified: Sat Jan 19 20:57:47 +0000 2008
### Description

```

A FAT32 or CD9660 filesystem can have directories that do not contain `..' entry (exactly why that happened, I'm not sure).  du(1) and find(1), which use fts(3), will exit with error message "fts_read: No such file or directory", once they 
encounter such directories.


```
### Audit trail

```


From: Jaromir Dolecek <jdolecek@NetBSD.org>
To: svs@ropnet.ru
Cc: gnats-bugs@gnats.netbsd.org
Subject: Re: admin/16651: fts(3), as used in base system, breaks on directories
 that do not contain `..' entry
Date: Sat, 4 May 2002 09:44:40 +0200 (CEST)

 Well, NetBSD msdosfs & cd9600 should fake '..' entry always.
 If not, it's a (different) bug. I.e. actual bug to be fixed
 is the kernel code faking the '..' entry, not dealing with it
 in userland.

 Can you describe in which case you get directory without '..'
 entry on NetBSD?

 Jaromir
 -- 
 Jaromir Dolecek <jdolecek@NetBSD.org> http://www.NetBSD.org/Ports/i386/ps2.html
 -=- We should be mindful of the potential goal, but as the tantric    -=-
 -=- Buddhist masters say, ``You may notice during meditation that you -=-
 -=- sometimes levitate or glow.   Do not let this distract you.''     -=-

From: Sergey Svishchev <svs@ropnet.ru>
To: Jaromir Dolecek <jdolecek@NetBSD.org>
Cc: gnats-bugs@gnats.netbsd.org
Subject: Re: admin/16651: fts(3), as used in base system, breaks on directories that do not contain `..' entry
Date: Sun, 5 May 2002 20:49:05 +0400

 On Sat, May 04, 2002 at 09:44:40AM +0200, Jaromir Dolecek wrote:
 > Can you describe in which case you get directory without '..'
 > entry on NetBSD?

 In cd9660 case, I don't have a test case handy (the disc is gone);  
 I'll try to recreate (i.e. damage :-) msdosfs in a day or two.

 Basically, this involves writing to the fs from NetBSD, then Win*, then
 again NetBSD...  add some scandisk runs to the mix... and you have it :)

 -- 
 Sergey Svishchev


Responsible-Changed-From-To: netbsd-admin->kern-bug-people 
Responsible-Changed-By: soren 
Responsible-Changed-When: Sat Aug 3 12:54:01 PDT 2002 
Responsible-Changed-Why:  
This is (at least) a kernel bug. 

```
